app/frontend/src/router/App.tsx

```diff
@@ -27,6 +28,7 @@ const router = createBrowserRouter(
         />
       </Route>
       <Route path="/demo" element={<Demo />} />
+      <Route path="*" element={<NotFoundPage />} />
```

Comment:
It think router v6 has a slightly different way of handling error elements. I have not investigated much into how it works or its pros and cons. Have you looked into it? Any thoughts?

Reply:
I took a hard look at the [resource from the issue](https://reactrouter.com/en/main/start/tutorial#not-found-data) and the RR [errorElement](https://reactrouter.com/en/main/route/error-element) but was unable to figure out how to use it to handle the 404.

My google searches for v6 handling of 404 kept coming back with using the wildcard
- https://denislistiadi.medium.com/react-router-v6-handling-404-pages-passing-props-to-link-da1ef843eb44
- https://stackoverflow.com/questions/67050966/how-to-build-a-404-page-with-react-router-dom-v6

I did just get this response from GPT4 that seems plausible
<details><summary>See screenshot</summary>

![image](https://user-images.githubusercontent.com/73561520/232252140-e36edc93-6883-43ff-947b-9bbf7d9b2a0f.png)


</details>

However, the docs do not mention it as far as I can tell and the [createRoutesFromElements](https://reactrouter.com/en/main/utils/create-routes-from-elements#createroutesfromelements) utility currently in use may not be compatible as it obfuscates away the `<Routes>` component


I'm not sure if it's worth investigating further unless the wildcard implementation has downsides I am unaware of?